lib/test/cdo/test_mailjet.rb

```diff
@@ -67,4 +67,31 @@ def test_send_template_email
 
     MailJet.send_template_email(to_email, to_name, email_config)
   end
+
+  def test_deletes_contact
+    email = 'fake.email@test.xx'
+
+    mock_contact = mock('Mailjet::Contact')
+    mock_contact.stubs(:id).returns(123)
+    Mailjet::Contact.expects(:find).with(email).returns(mock_contact)
+
+    mock_http_request = mock('Net::HTTP::Delete')
+    Net::HTTP::Delete.expects(:new).with(URI.parse("https://api.mailjet.com/v4/contacts/#{mock_contact.id}")).returns(mock_http_request)
+    mock_http_request.expects(:basic_auth).once
+
+    Net::HTTP.any_instance.expects(:request).with(mock_http_request)
+
+    MailJet.delete_contact(email)
```

Comment:
Nit: could there be another use of `Mailjet::Contact.expects(:find).with(email).returns(mock_contact)` after this last line but with `.returns(nil)` instead to confirm it was deleted?

Reply:
I'm not sure that would confirm it's actually deleted though, because we're not actually calling the MailJet service. Or am I misunderstanding?

Reply:
Oh gotcha, I thought maybe we could check it with that line to have the format of 1) check that it returns it like you'd expect, 2) delete it, 3) have the same check but expect it to be nil sort of deal. Totally fine if that wouldn't work!

Reply:
When I walk through those steps, I'm not sure what you're imagining for step 3. Can you maybe write some pseudocode?

For clarity, what I'm understanding is:

1. to change line 76 to be 
```
Mailjet::Contact.expects(:find).with(email).returns(mock_contact).then.returns(nil)
```

2. already done in line 84!

3. not sure

Reply:
Resolved offline: I misunderstood what the line was doing and this test makes sense!